lib/pubsub/iam.js

```diff
@@ -0,0 +1,225 @@
+ */
+function IAM(pubsub, resource) {
+  this.resource = resource;
+  this.makeReq_ = pubsub.makeReq_.bind(pubsub);
+}
+
+/**
+ * Get the IAM policy
+ *
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.policy - The [policy](https://cloud.google.com/pubsub/reference/rest/Shared.Types/Policy).
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @alias iam.getPolicy
+ *
+ * @resource [Topics: getIamPolicy API Documentation]{@link https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/getIamPolicy}
+ * @resource [Subscriptions: getIamPolicy API Documentation]{@link https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/getIamPolicy}
+ *
+ * @example
+ * topic.iam.getPolicy(function(err, policy, apiResponse) {});
+ *
+ * subscription.iam.getPolicy(function(err, policy, apiResponse) {});
+ */
+IAM.prototype.getPolicy = function(callback) {
+  var path = this.resource + ':getIamPolicy';
+
+  this.makeReq_('GET', path, null, null, function(err, resp) {
+    callback(err, resp, resp);
+  });
+};
+
+/**
+ * Set the IAM policy
+ *
+ * @throws {Error} If no policy is provided.
+ *
+ * @param {object} policy - The [policy](https://cloud.google.com/pubsub/reference/rest/Shared.Types/Policy).
+ * @param {array=} policy.bindings - Bindings associate members with roles.
+ * @param {object[]=} policy.rules - Rules to be applied to the policy.
+ * @param {string=} policy.etag - Etags are used to perform a read-modify-write.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object} callback.policy - The updated policy.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @alias iam.setPolicy
+ *
+ * @resource [Topics: setIamPolicy API Documentation]{@link https://cloud.google.com/pubsub/reference/rest/v1/projects.topics/setIamPolicy}
+ * @resource [Subscriptions: setIamPolicy API Documentation]{@link https://cloud.google.com/pubsub/reference/rest/v1/projects.subscriptions/setIamPolicy}
+ * @resource [Policy]{@link https://cloud.google.com/pubsub/reference/rest/Shared.Types/Policy}
+ *
+ * @example
+ * var myPolicy = {
+ *   bindings: [
+ *     {
+ *       role: 'roles/pubsub.subscriber',
+ *       members: ['serviceAccount:myotherproject@appspot.gserviceaccount.com']
+ *     }
+ *   ]
+ * };
+ *
+ * topic.iam.setPolicy(myPolicy, function(err, policy, apiResponse) {});
+ *
+ * subscription.iam.setPolicy(myPolicy, function(err, policy, apiResponse) {});
+ */
+IAM.prototype.setPolicy = function(policy, callback) {
+  if (!is.object(policy)) {
+    throw new Error('A policy is required');
+  }
+
+  var path = this.resource + ':setIamPolicy';
+  var body = {
+    policy: policy
+  };
+
+  this.makeReq_('POST', path, null, body, function(err, resp) {
+    callback(err, resp, resp);
```

Comment:
Sorry, just caught this. This needs error handling:

```js
if (err) {
  callback(err, null, resp);
  return;
}

callback(null, resp, resp);
```

Otherwise, a user's callback will get an error api response for `policy`.